common/src/Utilities/FolderUtils.cpp

```diff
@@ -0,0 +1,153 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2010  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "FolderUtils.h"
+#include <wx/file.h>
+#include <wx/stdpaths.h>
+#include <wx/filename.h>
+#include "ghc/filesystem.h"
+#include <wx/utils.h>
+
+// ---------------------------------------------------------------------------------
+//  Path namespace
+// ---------------------------------------------------------------------------------
+bool Path::IsRelative(const std::string &path)
+{
+    return fs::path(path).is_relative();
+}
+
+bool Path::IsDirectoryWithinDirectory(fs::path base, fs::path dir)
+{
+    fs::path relativePath = fs::relative(fs::absolute(dir), fs::absolute(base));
+    if (relativePath.empty())
+    {
+        return false;
+    }
+    return relativePath.string().rfind("..", 0) != 0;
+}
+
+s64 Path::GetFileSize(const fs::path &path)
+{
+    if (!fs::exists(path))
+        return -1;
+    return (s64)fs::file_size(path);
+}
+
+// Note std filesystem contains no clean way to normalize so we must use GUI provided functionality
+std::string Path::Normalize(const std::string &src)
+{
+    wxFileName normalize(Path::ToWxString(src));
+    normalize.Normalize();
+    return Path::FromWxString(normalize.GetFullPath());
+}
```

Comment:
If we truly need normalization, we should avoid wx and implement the following algorithm described in the docs instead https://en.cppreference.com/w/cpp/filesystem/path